### ISSUE TITLE
Failure to install on hassio

### ISSUE THREAD

**Issue by Louis-Demers** [TD22057/insteon-mqtt#139]
Hi, wanted to update without success. So I decided to uninstall and re-install. I cannot seem to get the add-on to appear in the add-on store... I get the following error message

```
19-04-04 20:44:03 INFO (MainThread) [hassio.addons.git] Update add-on https://github.com/home-assistant/hassio-addons repository
19-04-04 20:44:03 INFO (MainThread) [hassio.addons.git] Update add-on https://github.com/hassio-addons/repository repository
19-04-04 20:44:04 WARNING (MainThread) [hassio.addons.data] Can't read /data/addons/local/insteon-mqtt/config.json: required key not provided @ data['arch']. Got None
```

I tried to edit the config.json to replace {arch} by armhf or armv7 and got the exact same error message. 

I'm running the latest of everything on an rPi 3B+,  
homeassistant 0.91.0
hassio supervisor 152
hassos	2.11

**Comment by TD22057** [comment]
FYI - that's maintained by @lnr0626 right now so you may want to post this on his add-ons repo.

**Comment by lnr0626** [comment]
It looks like there was an update to the config file format. I'm pretty swamped right now (have been recently so haven't been able to play with this stuff for a while), but if no one gets to putting in a PR I'll give it a shot next time I'm free. Here's the relevant docs: https://developers.home-assistant.io/docs/en/hassio_addon_config.html

**Comment by Louis-Demers** [comment]
Thanks @TD22057  and @lnr0626  for the info. I know nothing abut docker and little about the rest. Here is how i fixed my problem based upon the link you provided. I edited the config.json to be 

```
{
    "name": "Insteon MQTT",
    "description": "Python Insteon PLM <-> MQTT bridge",
    "slug": "insteon-mqtt",
    "version": "0.6.8",
    "startup": "services",
    "arch": ["armhf"],
    "boot": "auto",
    "auto_uart": true,
    "map": ["config:rw"],
    "options": {},
    "schema": {},
    "image": "lnr0626/{arch}-insteon-mqtt"
}
```

i.e. I added the line` "arch": ["armhf"]` because I'm on a rPi 3B+. It didn't work because the latest tag available at  lnr0626 is 0.6.7 so  i edited version to 0.6.7.  It then worked and everything was working again.  I then "dared" to edit version back to 0.6.8 and change  image to  `"image": "td22057/{arch}-insteon-mqtt"` 
which also worked. This fix is probably not the proper way to do things because others may not be on rPi3B+ and would therefore need something else than armhf.... But at least its a way to get back up and running.

So my final config.json is 
```
{
    "name": "Insteon MQTT",
    "description": "Python Insteon PLM <-> MQTT bridge",
    "slug": "insteon-mqtt",
    "version": "0.6.8",
    "startup": "services",
    "arch": ["armhf"],
    "boot": "auto",
    "auto_uart": true,
    "map": ["config:rw"],
    "options": {},
    "schema": {},
    "image": "td22057/{arch}-insteon-mqtt"
}
```

Thanks for the clues.

**Comment by Louis-Demers** [comment]
Well, concluded too soon. Yes things did install properly, but only half worked. Pressing dimmer switches updates the state in home assistant which confirmed things were connected to mqtt and HA was accessing modem correctly. But using HA to turn on or OFF my insteon devices did not work.  Reinstalled again using this time 
```
{
    "name": "Insteon MQTT",
    "description": "Python Insteon PLM <-> MQTT bridge",
    "slug": "insteon-mqtt",
    "version": "0.6.7",
    "startup": "services",
    "arch": ["armhf"],
    "boot": "auto",
    "auto_uart": true,
    "map": ["config:rw"],
    "options": {},
    "schema": {},
    "image": "lnr0626/{arch}-insteon-mqtt"
}
```
worked and I seem to be fully operational again. So it would seem to be a problem with 0.6.8 or the image at td22057 ....

**Comment by TD22057** [comment]
@Louis-Demers  If there is a problem with 0.6.8, I'd love to get it fixed.  Do you have the logs from that version - it would help identify if there is a problem?

**Comment by Louis-Demers** [comment]
@TD22057  i'll be happy to reinstall 0.6.8 and gather any log you wish for. I imagine  there would be some log level i would need to configure for maximum information ?

Where would this log appear, in the hassio page and system tab and in the home assistant developper tools info pane ?

The behaviour was
1) mqtt/homeassistant would receive updates of the status of switches/dimmer/concats...
2) however when i try to use home assistant to turn on/off a light, noting happen and after a few seconds, it would revert the indicator to its original state.

PS: i also have questions about receiving battery updates, where can i ask such questions ? I feel bad about opening an issue for something that is most likely not a bug, but my misunderstanding....

**Comment by TD22057** [comment]
No idea - I don't run hassio so I don't know what happens with plugins.  I think they can be shown in the web front in somewhere.

Sounds like the command topics aren't working for some reason.  Not sure why that would be different between 6.7 to 6.8 but the log messages would probably make it pretty clear.

You can post in the HA forum (search for the insteon-mqtt thread) or just open a new ticket - either way is fine.

**Comment by Louis-Demers** [comment]
@TD22057 I finally updated everything, backup and used  the flooring
```
{
    "name": "Insteon MQTT",
    "description": "Python Insteon PLM <-> MQTT bridge",
    "slug": "insteon-mqtt",
    "version": "0.6.8",
    "startup": "services",
    "arch": ["armhf"],
    "boot": "auto",
    "auto_uart": true,
    "map": ["config:rw"],
    "options": {},
    "schema": {},
    "image": "td22057/{arch}-insteon-mqtt"
}
```
Updated inteon-mqtt without any problems and everything works fine !!!! So seems 0.6.8 is now working fine for me.  Thanks.

**Comment by TD22057** [comment]
PR#169 manually merged so this should be fixed.  New config will be available in 0.6.9 when it's out.

**Comment by JonS9999** [comment]
Ted - wow, you have been busy today. :-)
